extensions/data-workspace/src/dialogs/newProjectQuickpick.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as constants from '../common/constants';
+import { directoryExist } from '../common/utils';
+import { defaultProjectSaveLocation } from '../common/projectLocationHelper';
+import { WorkspaceService } from '../services/workspaceService';
+
+/**
+ * Create flow for a New Project using only VS Code-native APIs such as QuickPick
+ */
+export async function createNewProjectWithQuickpick(workspaceService: WorkspaceService): Promise<void> {
+	// Refresh list of project types
+	const projectTypes = (await workspaceService.getAllProjectTypes()).map(projType => {
+		return {
+			label: projType.displayName,
+			description: projType.description,
+			id: projType.id
+		} as vscode.QuickPickItem & { id: string };
+	});
+
+	// 1. Prompt for project type
+	const projectType = await vscode.window.showQuickPick(projectTypes, { title: constants.SelectProjectType, ignoreFocusOut: true });
+	if (!projectType) {
+		return;
+	}
+
+	// 2. Prompt for project name
+	const projectName = await vscode.window.showInputBox(
+		{
+			title: constants.EnterProjectName,
+			validateInput: (value) => {
+				return value ? undefined : constants.NameCannotBeEmpty;
+			},
+			ignoreFocusOut: true
+		});
+	if (!projectName) {
+		return;
+	}
+
+	// 3. Prompt for Project location
+	// Show quick pick with just browse option to give user context about what the file dialog is for (since that doesn't always have a title)
```

Comment:
Is this a common pattern?  Feels a bit weird, though I get what you're going for.

Reply:
No idea, as mentioned I plan on following up with folks on how it feels and looking at other similar flows to see if there's a common pattern. But jumping straight to the open dialog feels really unfriendly.